datafusion/functions/src/string/common.rs

```diff
@@ -103,6 +104,7 @@ pub(crate) fn general_trim<T: OffsetSizeTrait>(
 /// This function errors when:
 /// * the number of arguments is not 1
 /// * the first argument is not castable to a `GenericStringArray`
+#[allow(dead_code)]
```

Comment:
Are these functions still needed? Can we instead simply delete them if they are no longer used (since they are crate private)?

They don't appear to be used anywhere else: https://github.com/search?q=repo%3Aapache%2Farrow-datafusion%20unary_string_function&type=code

Reply:
You are right; this can be removed. I plan to delete these seemingly unnecessary pieces of code after they have been reviewed.